### ISSUE TITLE
Consolidate all page list screens to use dirPagination.

### ISSUE THREAD

**Issue by frankgiordano** [frankgiordano/PlatformTracker#71]
All the current list page screens either use a table grid or smart table. Both only support frontend pagination. As such, all the page results are retrieved from the backend. I need to change the list screens to have the same look and feel with a list that supports pagination on both ends. As a result, the usability becomes consistent and avoids performance issues when thousands of page objects exist, not all results are returned. 

This is the plugin I will be using to accomplish this task. 

https://github.com/michaelbromley/angularUtils

**Comment by frankgiordano** [comment]
I converted Product and Project list screens. Add pagination support on the backend that links to the frontend usage. I streamlined the code and did some clean up also.

**Comment by frankgiordano** [comment]
Onward with Root Cause and Resolutions list screens next.

**Comment by frankgiordano** [comment]
The only list page that I will leave as be a grid table list is the Link Resolutions page. This page if suited for gird table as I need check box processing. In addition, I do not see a reason this page needs pagination on the back end size for now.

**Comment by frankgiordano** [comment]
I made changes to the padding of the screens for top and bottom screen title so that it spaces out well with empty list screens and with this change I improvement each pages spacing and removed the following html I used previous to do spacing which was ugly <div style="clear:both;">&nbsp;</div>. I replaced and added many styles with padding top and bottom usage.

**Comment by frankgiordano** [comment]
Converted Root Cause list screen to use dirPagination.

**Comment by frankgiordano** [comment]
I also made changes to most create/edit screens so that the page title also disappears when progress bar is shown.

**Comment by frankgiordano** [comment]
Also in the last checkin I made changes to the attributes of the project object there were some typos and one variable I changed to use camel case like all others.

**Comment by frankgiordano** [comment]
Resolution list screen conversion is next.

**Comment by frankgiordano** [comment]
Resolution conversion completed. Onward to Incident.

**Comment by frankgiordano** [comment]
complete full conversion of the incident group list screen.

Incident list screen is next!

**Comment by frankgiordano** [comment]
Noticed with further testing Root Cause, Project, and Resolution search list screen did not keep track of its page and search result when returning from edit detail screen originated from selecting an item for the search list screen. I will implement "current-page" handling which is a new feature of dirPagination. I will upgrade to the latest dirPagination code base as such. For Incident Group and Product list screens this is not a problem as I am handling the search and edit screens in one html file using the ng-show etc to toggle between them and hence the search screen is always where it was last left.

**Comment by frankgiordano** [comment]
Done for previous comment's work. 

Onward with the main Incident search list screen conversion.

**Comment by frankgiordano** [comment]
Incident conversion completed.

In summary, following list screens converted Incident, Incident Group, Product, Project, Root Cause, and Resolution.

The following screens list I have decided to stay with the ngTable format which are Link Resolutions, Incidents From Group, and Weekly Incident Report. The first two are specialized screens and I think are best for ngTable at the moment even though no backend pagination is done. You will see why if you use the product. The latter I will need to eventually change. I will revisit it in the future. At the moment I will consider this issue closed.

Lastly, I added Create button in the Incident list screen for short cut creation as some other list pages have it and it is useful to have it there.
